key/src/main/java/net/kyori/adventure/key/KeyFormat.java

```diff
@@ -0,0 +1,249 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.key;
+
+import java.util.stream.Stream;
+import net.kyori.examination.Examinable;
+import net.kyori.examination.ExaminableProperty;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A factory to create {@link Key}s using configured defaults.
+ *
+ * <p>The configuration options include:</p>
+ * <dl>
+ *   <dt>namespace</dt>
+ *   <dd>The namespace to be used as fallback, when no other namespace is specified</dd>
+ *   <dt>separator</dt>
+ *   <dd>The character separating namespace and value</dd>
+ * </dl>
+ *
+ * <p>Example usage of this class:</p>
+ * <pre>
+ *   KeyFormat format = keyFormat("adventure", ':');
+ *   Key key = format.key("translations");
+ *   format.asString(key); // -> "adventure:translations"
+ * </pre>
+ *
+ * <pre>
+ *   KeyFormat format = keyFormat("adventure", '>');
+ *   Key key = format.key("translations");
+ *   format.asString(key); // -> "adventure>translations"
+ * </pre>
+ *
+ * @see Key#key(String)
+ * @since 4.22.0
+ */
+public interface KeyFormat extends Namespaced, Examinable {
+  /**
+   * Gets the default minecraft key format instance.
+   *
+   * <p>This uses {@link Key#MINECRAFT_NAMESPACE} and {@link Key#DEFAULT_SEPARATOR}.</p>
+   *
+   * @return the instance
+   * @since 4.22.0
+   */
+  static @NotNull KeyFormat minecraft() {
+    return KeyFormatImpl.MINECRAFT_INSTANCE;
+  }
+
+  /**
+   * Creates a key format with a fallback {@code namespace} and the {@link Key#DEFAULT_SEPARATOR}.
+   *
+   * @param namespace the namespace to be used as fallback in {@link #key(String)}
+   * @return the key format
+   * @throws IllegalArgumentException if the namespace contains an invalid character
+   * @since 4.22.0
+   */
+  static KeyFormat namespace(@KeyPattern.Namespace final @NotNull String namespace) {
+    return keyFormat(namespace, Key.DEFAULT_SEPARATOR);
+  }
+
+  /**
+   * Creates a key format with a {@code separator} and the default {@link Key#MINECRAFT_NAMESPACE}.
+   *
+   * @param separator the separator to be used when parsing keys using {@link #key(String)}
+   * @return the key format
+   * @since 4.22.0
+   */
+  static KeyFormat separator(final char separator) {
+    return keyFormat(Key.MINECRAFT_NAMESPACE, separator);
+  }
```

Comment:
This feels weird to have a method. I think just the methods with the namespace arg, and a method with the namespace arg and char separator, instead of implicitly using minecraft here.